src/Tabs.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+import core from 'metal';
+import templates from './Tabs.soy';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+
+/**
+ * UI Component to navigate through tabbed data.
+ */
+class Tabs extends Component {
+	/**
+	 * Adds a tab to the tabs array at the specified index if given or
+	 * appends it at the end.
+	 * @param {Object} tab
+	 * @param {number} index
+	 */
+	addTab(tab, index) {
```

Comment:
This may need a `this.tabs = this.tabs` at the end, similar to what you do in `setTabDisabled`, otherwise calling this won't rerender.